test/unit/modunit_pico_mld.c

```diff
@@ -118,16 +119,12 @@ END_TEST
 START_TEST(tc_pico_mld_report_expired)
 {
     struct mld_timer *t = PICO_ZALLOC(sizeof(struct mld_timer));
-    struct pico_ip6 zero = {{0}};
-    struct mcast_parameters p;
+    struct pico_ip6 zero = {0};
```

Comment:
actually, on my compiler this adds a new warning : 
```
test/unit/modunit_pico_mld.c:122:12: warning: missing braces around initializer [-Wmissing-braces]
     struct pico_ip6 zero = {0};
```
There are a few other places where you also have this.